### ISSUE TITLE
Add support for the type parameter to aws_spot_fleet_request

### ISSUE THREAD

**Issue by WheresWardy** [terraform-providers/terraform-provider-aws#4266]
Currently Terraform does not support the `Type` parameter when using the `aws_spot_fleet_request` resource.

The parameter as supported by the API (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_SpotFleetRequestConfigData.html) allows you to specify whether you want to just create spot instances according to the current request, or maintain the specified number of targets when the spot instances are terminated.

Because the default is `request`, currently you cannot create spot fleet requests that maintain the target number of spot instances using Terraform.

### Terraform Version
```
Terraform v0.11.7
+ provider.aws v1.15.0
+ provider.template v1.0.0
```

### Affected Resource(s)
- aws_spot_fleet_request

### Expected Behavior
Terraform should support the type parameter as supported by the AWS API for spot fleet requests.

### Actual Behavior
Terraform replies with an error saying that the `type` parameter is not supported.

**Comment by bflad** [comment]
The enhancement to specify fleet type is in master and will release with version 1.26.0 of the AWS provider, likely middle next week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.26.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1260-july-04-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
